From 1d6d869d498227d6f13b312a6a00ab31de5996a9 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Mon, 9 Jan 2012 14:34:24 -0500
Subject: [PATCH] build: Add --disable-documentation configure option

Documentation is the source of many cyclical build loops.  Allow
operating system builders to manually cut these cycles initially
by disabling documentation for a first pass.
---
 Makefile.am  |   10 ++++++++--
 configure.ac |    9 ++++++++-
 2 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 7a35395..08213ce 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -191,12 +191,15 @@ udev_udevadm_LDADD = libudev/libudev-private.la
 # ------------------------------------------------------------------------------
 # udev man pages
 # ------------------------------------------------------------------------------
-dist_man_MANS = \
+dist_man_MANS = 
+dist_noinst_DATA =
+if ENABLE_DOCUMENTATION
+dist_man_MANS += \
 	udev/udev.7 \
 	udev/udevadm.8 \
 	udev/udevd.8
 
-dist_noinst_DATA = \
+dist_noinst_DATA += \
 	udev/udev.html \
 	udev/udevadm.html \
 	udev/udevd.html
@@ -205,6 +208,7 @@ EXTRA_DIST += \
 	udev/udev.xml \
 	udev/udevadm.xml \
 	udev/udevd.xml
+endif
 
 udev/%.7 udev/%.8 : udev/%.xml
 	$(AM_V_GEN)$(XSLTPROC) -o $@ -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $<
@@ -270,7 +274,9 @@ extras_scsi_id_scsi_id_SOURCES =\
 	extras/scsi_id/scsi_id.h
 extras_scsi_id_scsi_id_LDADD = libudev/libudev-private.la
 libexec_PROGRAMS += extras/scsi_id/scsi_id
+if ENABLE_DOCUMENTATION
 dist_man_MANS += extras/scsi_id/scsi_id.8
+endif
 EXTRA_DIST += extras/scsi_id/README
 
 # ------------------------------------------------------------------------------
diff --git a/configure.ac b/configure.ac
index e387fac..3c7b830 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,7 +15,6 @@ LT_INIT([disable-static])
 AC_PROG_AWK
 GTK_DOC_CHECK(1.10)
 AC_PREFIX_DEFAULT([/usr])
-AC_PATH_PROG([XSLTPROC], [xsltproc])
 
 AC_SEARCH_LIBS([clock_gettime], [rt], [], [AC_MSG_ERROR([POSIX RT library not found])])
 
@@ -39,6 +38,14 @@ AS_IF([test "x$with_selinux" = "xyes"], [
 AC_SUBST([SELINUX_LIBS])
 AM_CONDITIONAL(WITH_SELINUX, [test "x$with_selinux" = "xyes"])
 
+AC_ARG_ENABLE([documentation],
+	AS_HELP_STRING([--enable-documentation], [enable man pages and HTML]),
+	[], [enable_documentation=yes])
+AM_CONDITIONAL(ENABLE_DOCUMENTATION, test x$enable_documentation = xyes)
+if test x$enable_documentation = xyes; then
+   AC_PATH_PROG([XSLTPROC], [xsltproc])
+fi
+
 AC_ARG_ENABLE([debug],
 	AS_HELP_STRING([--enable-debug], [enable debug messages @<:@default=disabled@:>@]),
 	[], [enable_debug=no])
-- 
1.7.6.4

